src/python_testing/TC_CADMIN_1_5.py

```diff
@@ -44,13 +45,59 @@
 
 
 class TC_CADMIN_1_5(MatterBaseTest):
-    async def get_txt_record(self):
+    async def get_all_txt_records(self):
         discovery = mdns_discovery.MdnsDiscovery(verbose_logging=True)
-        comm_service = await discovery.get_commissionable_service(
-            discovery_timeout_sec=240,
-            log_output=False,
-        )
-        return comm_service
+        discovery._service_types = [mdns_discovery.MdnsServiceType.COMMISSIONABLE.value]
+        await discovery._discover(discovery_timeout_sec=240, log_output=False)
+
+        if mdns_discovery.MdnsServiceType.COMMISSIONABLE.value in discovery._discovered_services:
+            return discovery._discovered_services[mdns_discovery.MdnsServiceType.COMMISSIONABLE.value]
+        return []
+
+    async def wait_for_correct_cm_value(self, expected_cm_value, expected_discriminator, max_attempts=5, delay_sec=5):
```

Comment:
```suggestion
    async def wait_for_correct_cm_value(self, expected_cm_value: int, expected_discriminator: int, max_attempts=5, delay_sec=5):
```

To avoid too much duck typing. Would this work?

Reply:
Thank you, this has been added when adding the dataclass to the code.